avail-server/src/main/kotlin/com/avail/server/io/WebSocketAdapter.kt

```diff
@@ -2049,5 +2069,32 @@ class WebSocketAdapter @Throws(IOException::class) constructor(
 			sendFrame(channel, Opcode.CLOSE, buffer)
 				{ channel.scheduleClose(CommunicationErrorDisconnect(it)) }
 		}
+
+		/**
+		 * Fail the WebSocket connection.
+		 *
+		 * @param channel
+		 *   A channel.
+		 * @param statusCode
+		 *   The [status code][WebSocketStatusCode].
+		 * @param reasonMessage
+		 *   The reason message.
+		 * @param reason
+		 *   The [DisconnectReason].
+		 */
+		private fun fail(
+			channel: WebSocketChannel,
+			statusCode: WebSocketStatusCode,
+			reasonMessage: String,
+			reason: DisconnectReason)
+		{
+			val utf8 = StandardCharsets.UTF_8.encode(reasonMessage)
+			val buffer = ByteBuffer.allocateDirect(utf8.limit() + 2)
+			buffer.putShort(statusCode.statusCode.toShort())
+			buffer.put(utf8)
+			channel.channelCloseHandler.reason = reason
+			sendFrame(channel, Opcode.CLOSE, buffer)
+			{ channel.scheduleClose(CommunicationErrorDisconnect(it)) }
```

Comment:
For safety, this probably needs to be reformatted:

```
sendFrame(channel, Opcode.CLOSE, buffer) {
	channel.scheduleClose(CommunicationErrorDisconnect(it))
}
```

Reply:
I made the requested change.